admin/assets/javascripts/global-scripts.js

```diff
@@ -227,70 +242,107 @@ $(function () {
     }
   }
 
-  /**
-   * Group filters methods.
-   * @type {{tableRowVisibility: tableRowVisibility, findActiveGroups: findActiveGroups, checkGroupCheckbox: checkGroupCheckbox, updateSortingLink: updateSortingLink}}
-   */
-  var groupFilters = {
+ /**
+  * Util methods to help manage the checkbox state.
+  * @type {{checkCheckbox: checkCheckbox, updateSortingLink: updateSortingLink, tableRowVisibility: tableRowVisibility, reselectPreviousValues: reselectPreviousValues, getQueryParam: getQueryParam}}
+  */
+  var checkboxUtil = {
+    /**
+     * Change checkbox status to 'checked' for passed `param`Ids.
+     * @param `param`Ids
+     * @returns {boolean}
+     */
+    checkCheckbox: function (param, paramIds) {
+      if (!paramIds) { return false }
+      $('input[name="' + param + '"]').prop('checked', false).attr('data-checked', false)
+      $(paramIds).each(function (key, value) {
+        $('input[id="' + param + '-' + value + '"]').prop('checked', true).attr('data-checked', true)
+      })
+    },
+
+    /**
+     * Update sorting link to include active parameters - group / pupil ids.
+     * @param action
+     * @param groupId
+     */
+    updateSortingLink: function (action, param, paramId) {
+      var sortingLinks = $('.sortingLink')
+      sortingLinks.each(function () {
+        var sortingLink = $(this)
+        var insertParamId = paramId + ','
+        if (sortingLink.length > 0) {
```

Comment:
@danwilliams any chance we can simplify this a bit?
My personal view is that it's not really that readable.